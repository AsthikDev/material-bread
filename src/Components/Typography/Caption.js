import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import BaseText from './BaseText';

const styles = StyleSheet.create({
  caption: {
    fontSize: 12,
    lineHeight: 12,
    letterSpacing: 0.4,
  },
});

export default class Caption extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    return (
      <BaseText typographyStyles={styles.caption} {...this.props}>
        {this.props.children}
      </BaseText>
    );
  }
}
