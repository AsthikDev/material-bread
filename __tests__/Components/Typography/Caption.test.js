import React from 'react';
import Caption from '../../../src/Components/Typography/Caption';

import renderer from 'react-test-renderer';

test('it renders Caption', () => {
  const tree = renderer.create(<Caption>Caption</Caption>).toJSON();
  expect(tree).toMatchSnapshot();
});
