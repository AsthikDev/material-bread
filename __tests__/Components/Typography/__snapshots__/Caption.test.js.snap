// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Caption 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
      "fontSize": 12,
      "letterSpacing": 0.4,
      "lineHeight": 12,
    }
  }
  typographyStyles={
    Object {
      "fontSize": 12,
      "letterSpacing": 0.4,
      "lineHeight": 12,
    }
  }
>
  Caption
</Text>
`;
